From a0afd8443430554ce00c6a3952e6be05f7c7d9ca Mon Sep 17 00:00:00 2001
From: amatej <amatej@redhat.com>
Date: Tue, 11 Dec 2018 12:44:21 +0100
Subject: [PATCH] Reset callback to NULL (RhBug:1637923)

It prevents crash of PackageKit
 #0 repo_mirrorlist_failure_cb at /usr/src/debug/libdnf-0.20.0-1.fc29.x86_64/libdnf/dnf-repo.cpp:1641
 #1 check_transfer_statuses at /usr/src/debug/librepo-1.9.1-1.fc29.x86_64/librepo/downloader.c:1718
 #2 lr_perform at /usr/src/debug/librepo-1.9.1-1.fc29.x86_64/librepo/downloader.c:1956
 #3 lr_download at /usr/src/debug/librepo-1.9.1-1.fc29.x86_64/librepo/downloader.c:2083
 #4 lr_download_target at /usr/src/debug/librepo-1.9.1-1.fc29.x86_64/librepo/downloader.c:2187
 #5 lr_yum_download_url at /usr/src/debug/librepo-1.9.1-1.fc29.x86_64/librepo/yum.c:491
 #6 lr_download_url at /usr/src/debug/librepo-1.9.1-1.fc29.x86_64/librepo/downloader.c:2197
 #7 dnf_repo_download_import_public_key at /usr/src/debug/libdnf-0.20.0-1.fc29.x86_64/libdnf/dnf-repo.cpp:1616
 #8 dnf_repo_update at /usr/src/debug/libdnf-0.20.0-1.fc29.x86_64/libdnf/dnf-repo.cpp:1758
 #9 dnf_sack_add_repo at /usr/src/debug/libdnf-0.20.0-1.fc29.x86_64/libdnf/dnf-sack.cpp:2086
---
 libdnf/dnf-repo.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libdnf/dnf-repo.cpp b/libdnf/dnf-repo.cpp
index a96ce1c7d..1cf762b19 100644
--- a/libdnf/dnf-repo.cpp
+++ b/libdnf/dnf-repo.cpp
@@ -1906,6 +1906,7 @@ dnf_repo_update(DnfRepo *repo,
     g_free(updatedata.last_mirror_url);
     dnf_state_release_locks(state);
     lr_handle_setopt(priv->repo_handle, NULL, LRO_PROGRESSCB, NULL);
+    lr_handle_setopt(priv->repo_handle, NULL, LRO_HMFCB, NULL);
     lr_handle_setopt(priv->repo_handle, NULL, LRO_PROGRESSDATA, 0xdeadbeef);
     return ret;
 }
